frontends/common/parser_options.cpp

```diff
@@ -490,8 +490,8 @@ bool ParserOptions::isAnnotationDisabled(const IR::Annotation *a) const {
 }
 
 DebugHook ParserOptions::getDebugHook() const {
-    using namespace std::placeholders;
-    auto dp = std::bind(&ParserOptions::dumpPass, this, _1, _2, _3, _4);
+    auto dp = std::bind(&ParserOptions::dumpPass, this, std::placeholders::_1,
+                        std::placeholders::_2, std::placeholders::_3, std::placeholders::_4);
     return dp;
```

Comment:
As far as I understand it, the original code was actually using placeholders the way they were intended. There is basically nothing wrong with *local* `using namespace`.

Reply:
For bf-p4c, I had to make the namespace explicit because the unified builds also ended up including boost placeholders. This will lead to a conflict. It seems to be an unfortunate side-effect of including boost.

Reply:
Maybe someone is using `using namespace boost::placeholders` globally / on .cpp level? That would be the actual bug... (or not exactly bug, but in my opinion that would be the wrong use of the using in presence of something like unity build that breaks file scopes).

Reply:
I believe [this](https://www.boost.org/doc/libs/1_71_0/boost/bind.hpp) file is getting pulled in somewhere. 

Related: 
https://stackoverflow.com/questions/53203970/why-boostbind-insists-pulling-boostplaceholders-into-global-namespace

Reply:
We have discussed this internally, the problem is some other boost headers pull the `using namespace boost::placeholders` into global namespace (at least in older boost versions). Therefore this cannot be fixed nicely. In order to allow inclusion of such boost headers (e.g. `<boost/graph/transitive_closure.hpp>` for boost versions < 1.77) we need to keep the uses of `std::placeholders::_1` fully qualified). Thanks @fruffy for tracking the source.